homeassistant/components/google_generative_ai_conversation/conversation.py

```diff
@@ -326,41 +312,39 @@ async def async_process(
                         f"Sorry, I had a problem talking to Google Generative AI: {err}"
                     )
 
-                result.response.async_set_error(
-                    intent.IntentResponseErrorCode.UNKNOWN,
-                    error,
-                )
-                return result
+                raise HomeAssistantError(error) from err
 
             LOGGER.debug("Response: %s", chat_response.parts)
             if not chat_response.parts:
-                result.response.async_set_error(
-                    intent.IntentResponseErrorCode.UNKNOWN,
-                    "Sorry, I had a problem getting a response from Google Generative AI.",
+                raise HomeAssistantError(
+                    "Sorry, I had a problem getting a response from Google Generative AI."
                 )
-                return result
-            self.history[result.conversation_id] = chat.history
+            content = " ".join(
+                [part.text.strip() for part in chat_response.parts if part.text]
+            )
+            if content:
+                session.async_add_message(
+                    conversation.Content(
+                        role="assistant",
```

Comment:
Why not "model" to simplify the code in _chat_message_convert?

Reply:
`conversation.Content` is a standardized model that uses `assistant` as the string

![CleanShot 2025-01-29 at 09 32 14@2x](https://github.com/user-attachments/assets/3e52c418-8c56-47ea-8230-4c628433bc5c)

(from conversation/session.py)